Explorer/Assets/Scripts/ECS/StreamableLoading/AssetBundles/LoadAssetBundleSystem.cs

```diff
@@ -134,11 +134,21 @@ public static async UniTask<StreamableLoadingResult<AssetBundleData>> CreateAsse
             ReportData reportCategory,
             string version,
             string source,
+            bool lookForShaderAssets,
             CancellationToken ct)
         {
             // if the type was not specified don't load any assets
             if (expectedObjType == null)
                 return new StreamableLoadingResult<AssetBundleData>(new AssetBundleData(assetBundle, metrics, dependencies));
+            
+            if (expectedObjType == typeof(GameObject) && lookForShaderAssets)
```

Comment:
Perhaps you can invert some conditions since type-checking seems to be the most expensive operation, by either: 
```
if (lookForShaderAssets && expectedObjType == typeof(GameObject))
```
Or bring dependencies.length checking before this check

Reply:
I like it. Changing it